specification/hugr.md

```diff
@@ -980,97 +989,45 @@ indices after the list of node indices?
+There are three classes of type: Any $\supset$ Copyable $\supset$ Equatable. Types in these classes  are distinguished by the operations possible on (runtime) values of those types:
+  - For the broadest class ("Any" type), the only operation supported is the identity operation (aka no-op, or `lift` - see [Resource Tracking](#resource-tracking) below). Specifically, we do not require it to be possible to copy or discard all values, hence the requirement that outports of linear type must have exactly one edge. (That is, a type not known to be in the copyable subset). All incoming ports must have exactly one edge.
+
+    In fully qubit-counted contexts programs take in a number of qubits as input and return the same number, with no discarding. See [quantum resource](#quantum-resource) for more.
+
+  - The next class are "Copyable" types, aka "Classic" types, where values can be copied (and discarded, the 0-ary copy). This allows multiple (or 0) outgoing edges from an outport; also these types can be sent down Static edges.
 
-  - User-defined types
-  - Resource management - knowing what plugins a given graph depends on
+  - The final class are "Equatable" types: these are Copyable types where additionally there is a well-defined notion of equality between values. (While *some* notion of equality is defined on
+  any type with a binary representation, that if the bits are equal then the value is, the converse is not necessarily true - values that are indistinguishable can have different bit representations.)
 
-A grammar of available types is defined below.
+For example, a `float` type (defined in an extension) would be Copyable, but not Equatable. Also, Hugr "classes" loosely correspond to Tierkreis' notion of "constraints".
+
+**Row Types** The `#` is a *row type* which consists of zero or more types. Types in the row can optionally be given names in metadata i.e. this does not affect behaviour of the HUGR.
+
+The Hugr defines a number of type constructors, that can be instantiated into types by providing some collection of types as arguments. The constructors are given in the following grammar:
 
 ```haskell
-Type ::= [Resources]SimpleType
--- Rows are ordered lists, not sets
-#    ::= #(SimpleType)
-#(T) ::= (T)*
-
-Resources ::= (Resource)* -- set not list
-
-Container(T) ::= Tuple(#(T))
-              | Array<u64>(T)
-              | NewType(Name, T)
-              | Sum (#(T))
-ClassicType ::= USize
-              | Var(X)
-              | String
-              | Graph[R](#, #)
-              | Opaque(Name, #)
-              | Container(ClassicType)
-SimpleType ::= Qubit
-              | QPaque(Name, #)
-              | Container(SimpleType)  -- Sometimes called 'Qontainer'
-              | ClassicType
+
+Resources ::= (Resource)* -- a set, not a list
+
+Type ::= Tuple(#) -- fixed-arity, heterogenous components 
+       | Sum(#)   -- disjoint union of other types, ??tagged by unsigned int??
```

Comment:
this `tagged by unsigned int` was in here (a bit further down) before, and relates to the implementation and/or the `tag` operation, defined elsewhere. So this might want rewording. `can be tagged by USize` perhaps? or remove it, either with an implementation note elsewhere, or in whatever part of the prelude that defines the `tag` op ?